app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt

```diff
@@ -88,21 +102,31 @@ open class UserViewModel(
         if (firebaseUser != null) {
           // User is signed in, start listening to user data
           loadUser(firebaseUser.uid, firebaseUser)
-          // Fetch sent friend requests
+          // Listen to friend requests (incoming)
+          listenToFriendRequests()
+          // Listen to friend requests (sent)
           getSentFriendRequests()
         } else {
-          // User is signed out, clear user data and remove listeners
+          // User is signed out, clear data and remove listeners
           userListenerRegistration?.remove()
           userListenerRegistration = null
           _user.value = null
           _isLoading.value = false
+          sentFriendRequestsListener?.remove()
```

Comment:
I don't understand the need for the sentFriendRequestListener. I am operating under the assumption that the user is only logged in from one device at a time, so the only way he can send a friend request is from the add so maybe there is no use for this.

Reply:
Thank you for the feedback! Your assumption about single-device usage is valid in certain scenarios, but the sentFriendRequestListener serves a broader purpose. Let me clarify:
Although we may assume single-device usage, Firebase allows users to log in on multiple devices simultaneously. If a friend request is sent from another device, the current device would be unaware of this action without real-time updates. The listener ensures that such actions are reflected immediately.
Also, even if we maintain the assumption of single-device usage, the listener provides real-time updates to the UI. This eliminates the need to fetch the sent requests list repeatedly, improving user experience and reducing server load.
Even more, there is the issue of edge cases like: Network issues that might cause delayed requests or retries, and without the listener, these might not be reflected until a manual refresh. 
Removing this listener might simplify the code but at the cost of user experience and potential desynchronization across devices or sessions.